src/shared/edit-util.c

```diff
@@ -265,3 +346,53 @@ int trim_edit_markers(const char *path, const char *marker_start, const char *ma
 
         return 1; /* Changed, but good */
 }
+
+int do_edit_files_and_install(EditFileContext *context) {
+        int r;
+
+        assert(context);
+
+        if (context->n_files == 0)
+                return log_debug_errno(SYNTHETIC_ERRNO(ENOENT), "Got no files to edit");
+
+        EDIT_FILES_FOREACH(i, context)
+                if (isempty(i->temp)) {
+                        r = create_edit_temp_file(i->path,
+                                                  i->original_path,
+                                                  i->comment_paths,
+                                                  context->marker_start,
+                                                  context->marker_end,
+                                                  &i->temp,
+                                                  &i->line);
+                        if (r < 0)
+                                goto end;
+                }
+
+        r = run_editor(context);
+        if (r < 0)
+                goto end;
+
+        EDIT_FILES_FOREACH(i, context) {
+                /* Always call trim_edit_markers to tell if the temp file is empty */
+                r = trim_edit_markers(i->temp, context->marker_start, context->marker_end);
+                if (r < 0)
+                        goto end;
+                if (r == 0)
+                        continue;
+
+                r = RET_NERRNO(rename(i->temp, i->path));
+                if (r < 0) {
+                        log_error_errno(r, "Failed to rename \"%s\" to \"%s\": %m", i->temp, i->path);
+                        goto end;
+                }
+                i->temp = mfree(i->temp);
+
+                log_info("Successfully installed edited file '%s'.", i->path);
+        }
+
+        r = 0;
+
+end:
+        edit_file_context_unlink_all(context);
```

Comment:
This is not necessary, as the temporary files and unnecessary directories will be removed in `edit_file_context_done()` called in `verb_edit()`.